src/cargo/util/toml/mod.rs

```diff
@@ -2334,16 +2344,30 @@ impl TomlManifest {
 
 fn inheritable_from_path(
     config: &Config,
-    resolved_path: PathBuf,
+    workspace_path: PathBuf,
 ) -> CargoResult<InheritableFields> {
-    let key = resolved_path.parent().unwrap();
-    let source_id = SourceId::for_path(key)?;
-    let (man, _) = read_manifest(&resolved_path, source_id, config)?;
+    // Workspace path should have Cargo.toml at the end
+    let workspace_path_root = workspace_path.parent().unwrap();
+
+    // Let the barrow exit scope so that it can be picked up if there is a need to
```

Comment:
Is this a typo?

```suggestion
    // Let the borrow exit scope so that it can be picked up if there is a need to
```

Reply:
I've made this change in the latest commit